compendium/DeclarativeServices/src/ComponentContextImpl.cpp

```diff
@@ -69,50 +69,56 @@ namespace cppmicroservices
-                                }
-                                serviceMap.push_back(svc);
-                            }
-                            else
-                            {
-                                cppmicroservices::ServiceObjects<void> sObjs = bc.GetServiceObjects(sRefU);
-                                auto interfaceMap = sObjs.GetService();
-                                if (interfaceMap)
-                                {
-                                    foundAtLeastOneValidBoundService = true;
-                                    serviceMap.push_back(interfaceMap);
-                                }
-                            }
-                        }
-                    });
+                std::for_each(sRefs.rbegin(),
+                              sRefs.rend(),
+                              [&](cppmicroservices::ServiceReferenceBase const& sRef)
+                              {
+                                  if (sRef)
+                                  {
+                                      ServiceReferenceU sRefU(sRef);
+                                      auto refScope
+                                          = any_cast<std::string>(sRefU.GetProperty(Constants::SERVICE_SCOPE));
+                                      if (refScope.empty())
+                                      {
+                                          refScope = Constants::SCOPE_SINGLETON;
+                                      }
+                                      auto bc = GetBundleContext();
+                                      auto boundServicesCacheHandle = boundServicesCache.lock();
+                                      auto& serviceMap = (*boundServicesCacheHandle)[refName];
+                                      if (refScope == cppmicroservices::Constants::SCOPE_SINGLETON)
+                                      {
+                                          auto svc = bc.GetService(sRefU);
+                                          if (svc)
+                                          {
+                                              foundAtLeastOneValidBoundService = true;
+                                          }
+                                          serviceMap.push_back(svc);
+                                      }
+                                      else
+                                      {
+                                          cppmicroservices::ServiceObjects<void> sObjs = bc.GetServiceObjects(sRefU);
+                                          auto interfaceMap = sObjs.GetService();
+                                          if (interfaceMap)
+                                          {
+                                              foundAtLeastOneValidBoundService = true;
+                                              serviceMap.push_back(interfaceMap);
+                                          }
+                                      }
```

Comment:
I think this entire block can be simplified significantly and that the reference scope of the service reference doesn't need to be queried at all based off of the documentation for `ServiceObjects` class. `ServiceObjects` should do the right thing already based on the scope of the service.